### ISSUE TITLE
[CORE-901] Medusa js client missing order parameter

### ISSUE THREAD

**Issue by durdenx** [medusajs/medusa#2657]
### Describe the bug

The order option doesn't exist on the medusa js client for some endpoints (products.list in this case) :

```ts
const medusaClient = new Medusa({
  baseUrl: config.medusa.baseUrl,
  apiKey: config.medusa.secretKey,
  maxRetries: 3,
});

medusaClient.admin.products.list({
    offset: 0,
    limit: 20,
    order: "-title" // this is not working
  });
```

Could you regenerate the client or something?

### System information

Medusa version (including plugins):

- "@medusajs/medusa": "^1.6.4",
- "@medusajs/medusa-js": "^1.3.3",
- "@medusajs/medusa-cli": "^1.3.4",
- "medusa-fulfillment-manual": "^1.1.31",
- "medusa-interfaces": "^1.3.3",
- "medusa-payment-manual": "^1.0.16",
- "medusa-payment-stripe": "^1.1.47",
- "typeorm": "^0.2.45"
- SALES CHANNELS enabled

Node.js version: 18.12.1
Database: postgres 13.8
Operating system: MacOS 12.6.1
Browser (if relevant): 107.0.5304.110 

### Steps to reproduce the behavior

Use the medusaClient with the `order` parameter.

### Expected behavior

The `order` parameter is available in all list endpoints (especially the `admin.products.list` one) of the medusajs client and the client returns a sorted result with the provided parameter.

### Bonus!

I don't know if this is intended, but the client wants you to provide all the list type fields when updating: 

```ts
await medusaClient.admin.products.update(productId,
{
  collection_id: collectionId,

  // Needed to fix type issues
  // If we provide an empty array, the field value is overwritten.
  images: undefined as any,
  sales_channels: undefined as any,
}
```

### Context

- https://github.com/medusajs/medusa/discussions/2624

<sub>[CORE-901](https://linear.app/medusajs/issue/CORE-901/medusa-js-client-missing-order-parameter)</sub>

**Comment by olivermrbl** [comment]
Thanks for reporting this. We will attend to it in the near future 💪

**Comment by edihasaj** [comment]
any news on this?

**Comment by olivermrbl** [comment]
@edihasaj Solved in #2815. Did you test it yet?

**Comment by bkvaiude** [comment]
@olivermrbl I'm looking for the solution where I can sort by multiple columns

For example:

`Select * from products order by discounts, created_at desc`

in query params

`store/products?size=XL&order[]=discounts&order[]=-created_at`

It will helpful if guide how to solve this problem

I also looked into extending the `StoreGetProductsPaginationParams`
